src/_imaging.c

```diff
@@ -1571,21 +1573,46 @@ if (PySequence_Check(op)) { \
                 PyErr_SetString(PyExc_TypeError, must_be_sequence);
                 return NULL;
             }
-            int endian = strncmp(image->mode, "I;16", 4) == 0 ? (strcmp(image->mode, "I;16B") == 0 ? 2 : 1) : 0;
             double value;
-            for (i = x = y = 0; i < n; i++) {
-                set_value_to_item(seq, i);
-                if (scale != 1.0 || offset != 0.0) {
-                    value = value * scale + offset;
+            if (image->bands == 1) {
+                int bigendian;
+                if (image->type == IMAGING_TYPE_SPECIAL) {
+                    // I;16*
+                    bigendian = strcmp(image->mode, "I;16B") == 0;
                 }
-                if (endian == 0) {
-                    image->image8[y][x] = (UINT8)CLIP8(value);
-                } else {
-                    image->image8[y][x * 2 + (endian == 2 ? 1 : 0)] = CLIP8((int)value % 256);
-                    image->image8[y][x * 2 + (endian == 2 ? 0 : 1)] = CLIP8((int)value >> 8);
+                for (i = x = y = 0; i < n; i++) {
+                    set_value_to_item(seq, i);
+                    if (scale != 1.0 || offset != 0.0) {
+                        value = value * scale + offset;
+                    }
+                    if (image->type == IMAGING_TYPE_SPECIAL) {
+                        image->image8[y][x * 2 + (bigendian ? 1 : 0)] = CLIP8((int)value % 256);
+                        image->image8[y][x * 2 + (bigendian ? 0 : 1)] = CLIP8((int)value >> 8);
+                    } else {
+                        image->image8[y][x] = (UINT8)CLIP8(value);
+                    }
+                    if (++x >= (int)image->xsize) {
+                        x = 0, y++;
+                    }
                 }
-                if (++x >= (int)image->xsize) {
-                    x = 0, y++;
+            } else {
+                // BGR;*
+                int b;
+                for (i = x = y = 0; i < n; i++) {
+                    char ink[4];
+
+                    op = PySequence_Fast_GET_ITEM(seq, i);
+                    if (!op || !getink(op, image, ink)) {
+                        Py_DECREF(seq);
+                        return NULL;
+                    }
+                    /* FIXME: what about scale and offset? */
```

Comment:
This is copying an existing FIXME
https://github.com/python-pillow/Pillow/blob/07623d1a7cc65206a5355fba2ae256550bfcaba6/src/_imaging.c#L1639